1) Open the preferences in a tab instead of dialog for better UX. #6743

adityatoshniwal opened the pull request:
2) Add a search box to allow searching in the preferences tab. #2864

akshay-joshi commented on web/pgadmin/static/js/PgTreeView/index.jsx:
```diff
@@ -3,37 +3,64 @@ import { styled } from '@mui/material/styles';
 import EmptyPanelMessage from '../components/EmptyPanelMessage';
 import CheckBoxIcon from '@mui/icons-material/CheckBox';
-
+import useResizeObserver from 'use-resize-observer';
```
Add a copyright header to this file.

adityatoshniwal commented on docs/en_US/images/preferences_misc_user_interface.png:
This will be taken care in full preferences docs review.

adityatoshniwal commented:
> **GUI Related Comments**:
> 
> 1. When a workspace other than the default is selected, clicking the Settings button or selecting Preferences from the menu opens the Preferences tab in the default workspace. However, the active workspace does not switch accordingly.

Fixed.

> 2. Open the Preferences tab, make some changes, and click Reset All Preferences. Then select Save & Reload Later. While the changes are correctly applied in the backend, they are not reflected in the currently open tab. A manual refresh or closing and reopening of the Preferences tab is needed to see the updates.

Fixed. 

> 3. In the Preferences tab, go to Browser → Display, change any one toggle setting, and click Save. The setting is saved successfully. However, after this, attempting to toggle any controls again (especially multiple times) will not work; the controls become unresponsive.

Fixed.

> 4. Select the Browser node. You will see navigation links on the right side. Search for a string like "Layout", and click on any navigation link, it will fail, which is expected. Can we do something?

I have changed logic to select the first item automatically if filter changes.

> 
> **Source Code Related Comments**:
> 
> 1. Please update the missing preferences in the "preferences.rst" file, as you have already updated the screenshots. A few preferences are missing from the documentation (not related to your PR, but it's good to have)

As discussed, it will be taken care in a separate ticket.

> 2. Fix 5 old SonarQube code smells in Button.jsx if possible.

Fixed 3/5. Cognitive complexity didn't fix.
